util/progress/progressui/display.go

```diff
@@ -1056,6 +1056,7 @@ func (disp *ttyDisplay) print(d displayInfo, width, height int, all bool) {
 		lineCount++
 		if j.showTerm {
 			term := j.vertex.term
+			termHeight = max(termHeightMin, min(termHeight, height-termHeightMin-1))
```

Comment:
Shouldn't this be a new scoped variable, as I think previous computation shouldn't change the calculation for the next round.

Reply:
Good point that every round having the same result.  `termHeight` is used in both `print` and `update` though, so should it be a package-level variable?  Then I'd change it to something like:

```
const termHeightMin
var termHeightInitial
var termHeight
```
?

Reply:
I'm not sure you get what would be the difference between `termHeightInitial` and `termHeightMin`. If you need another var that sgtm though.

Reply:
Sure, here's what I was thinking:

`termHeightMin` is a const (and set to 6).
`termHeightInitial` is set to `BUILDKIT_TTY_LOG_LINES` if set (otherwise defaults to `termHeightMin`).

We then check bounds on `termHeightInitial` - if it passes, we set `termHeight` to it, otherwise it gets set to the larger of `termHeightMin` or `height-7`.  This way, the computation doesn't depend on the previous value for `termHeight`.

BTW I appreciate the many looks as I fumble around on this!